components/safelink/Editor.tsx

```diff
@@ -0,0 +1,312 @@
+'use client'
+import { ToolsOzoneSafelinkDefs } from '@atproto/api'
+import { FormEvent, useState, useEffect } from 'react'
+import {
+  useSafelinkAdd,
+  useSafelinkUpdate,
+  useSafelinkRules,
+} from './useSafelinkRules'
+import { useQueryClient } from '@tanstack/react-query'
+import { useLabelerAgent } from '@/shell/ConfigurationContext'
+import {
+  ActionTypeNames,
+  createSafelinkPageLink,
+  PatternTypeNames,
+  ReasonTypeNames,
+} from './helpers'
+import { Alert } from '@/common/Alert'
+import { ActionButton, LinkButton } from '@/common/buttons'
+import { Card } from '@/common/Card'
+import { FormLabel, Input, Select, Textarea } from '@/common/forms'
+import { ArrowLeftIcon } from '@heroicons/react/24/solid'
+import Link from 'next/link'
+import { useRouter, useSearchParams } from 'next/navigation'
+
+const getSubmitButtonText = (
+  rule: ToolsOzoneSafelinkDefs.UrlRule | null | undefined,
+  isSubmitting: boolean,
+) => {
+  if (!isSubmitting) {
+    return !!rule ? 'Update Rule' : 'Add Rule'
+  }
+  return !!rule ? 'Updating Rule...' : 'Adding Rule...'
```

Comment:
I don't think the casting of `rule` to a bool is necessary, but all good either way.